File: app/src/main/java/com/kieran/winnipegbusbackend/TripPlanner/classes/Monument.java
package com.kieran.winnipegbusbackend.TripPlanner.classes;

import org.json.JSONException;
import org.json.JSONObject;

import java.util.Locale;

public class Monument extends Address {
    int key;
    public Monument(JSONObject location) throws JSONException {
        super(location.getJSONObject("address"));

        key = location.getInt("key");
        title = location.getString("name");
    }

    @Override
    public String getURLString() {
        return String.format(Locale.CANADA, "monuments/%d", key);
    }
}

File: app/src/main/java/com/kieran/winnipegbusbackend/TripPlanner/classes/Monument.kt
package com.kieran.winnipegbusbackend.TripPlanner.classes

import org.json.JSONException
import org.json.JSONObject

import java.util.Locale

class Monument @Throws(JSONException::class)
constructor(location: JSONObject) : Address(location.getJSONObject("address")) {
    internal var key: Int = 0

    override val urlString: String
        get() = String.format(Locale.CANADA, "monuments/%d", key)

    init {

        key = location.getInt("key")
        title = location.getString("name")
    }
}
